src/Uno.UI/UI/Xaml/FrameworkTemplatePool.cs

```diff
@@ -240,9 +240,12 @@ private void OnParentChanged(object instance, object key, DependencyObjectParent
 
 		internal static void PropagateOnTemplateReused(object instance)
 		{
-			if (instance is IFrameworkTemplatePoolAware a)
+			if (instance is IFrameworkTemplatePoolAware a && instance is FrameworkElement b)
 			{
-				a.OnTemplateRecycled();
+				if (b.DataContext == null)
```

Comment:
```suggestion
				// If DataContext is not null, it means it has been explicitly set (not inherited). Resetting the view could push an invalid value through 2-way binding in this case.
				if (b.DataContext == null)
```

Reply:
This is a case that may need to be tested by a unit test. Also, testing for null may require determining the precedence of the value.

Reply:
Agreed for the test. 

@jeromelaban  Can you elaborate on why we would test for precedence in this case? At this point the template is detached from the visual tree, so if `DataContext` is non-null then we know it has been explicitly set.